src/python/pants/notes/2.9.x.md

```diff
@@ -1,5 +1,41 @@
 # 2.9.x Release Series
 
+## 2.9.0.dev4 (Dec 13, 2021)
+
+### New Features
+
+* scala: add `scalafmt` support for `fmt` and `lint` goals ([#13814](https://github.com/pantsbuild/pants/pull/13814))
+
+* scala: add protobuf codegen support via `scalapb` ([#13853](https://github.com/pantsbuild/pants/pull/13853))
+
+### User API Changes
+
+* Do not log extra deprecation info when there are no changes for `update-build-files --check`. ([#13855](https://github.com/pantsbuild/pants/pull/13855))
```

Comment:
Should probably put in `documentation`, which we've also been treating kind of like "UX-only changes"